ir/src/main/scala/org/scalajs/core/ir/Types.scala

```diff
@@ -84,13 +84,14 @@ object Types {
   case object NullType extends Type
 
   /** Reference types (allowed for classOf[], is/asInstanceOf[]). */
-  sealed abstract class ReferenceType extends Type
+  sealed trait ReferenceType
```

Comment:
This whole business is super hacky. We should definitely re-design this aspect of the IR for the next major version. At least with this PR we make it obvious that it's hacky ...